netbox/extras/views.py

```diff
@@ -729,21 +729,28 @@ def get_extra_context(self, request, instance):
             prechange_data = instance.prechange_data
 
         if prechange_data and instance.postchange_data:
-            diff_added = shallow_compare_dict(
-                prechange_data or dict(),
-                instance.postchange_data or dict(),
-                exclude=['last_updated'],
-            )
-            diff_removed = {
-                x: prechange_data.get(x) for x in diff_added
-            } if prechange_data else {}
+            diff_added, diff_removed = deep_compare_dict(prechange_data, instance.postchange_data, exclude=('last_updated'))
+            custom_fields_added = diff_added['custom_fields'] if 'custom_fields' in diff_added else None
+            custom_fields_removed = diff_removed['custom_fields'] if 'custom_fields' in diff_removed else None
+            cfr_list = []
+            if custom_fields_added:
+                for cf, cf_value in prechange_data['custom_fields'].items():
+                    cfr_list.append((cf, cf_value, cf in custom_fields_added))
+            cfa_list = []
+            if custom_fields_removed:
+                for cf, cf_value in instance.postchange_data['custom_fields'].items():
+                    cfa_list.append((cf, cf_value, cf in custom_fields_removed))
```

Comment:
I'm afraid this sort of approach isn't going to be maintainable long-term. The diffing implementation needs to operate generically; we can't give custom fields specifically any special treatment. (Consider what changes would be needed to support some similarly structured field in a future release.) Ultimately, we want a solution that can follow nested data structures and roll back up their diffs, without any context about the data contained within.